### PR TITLE
update the SemanticSegmentationTranslator

### PR DESCRIPTION
**Pull request by wxm2018**
## Description ##

Brief description of what this PR is about

update the SemanticSegmentationTranslator to get the original photo pixels

### CODE REVIEW

File: api/src/main/java/ai/djl/modality/cv/output/Segmentation.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+public class Segmentation implements JsonSerializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final Gson GSON =
+            JsonUtils.builder()
+                    .registerTypeAdapter(Segmentation.class, new SegmentationSerializer())
+                    .create();
+
+    private int[][] mask;
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/modality/cv/output/Segmentation.java:40]: Segmentation should hold classNames

File: api/src/main/java/ai/djl/modality/cv/translator/SemanticSegmentationTranslator.java [hunk 2]
```diff
@@ -13,32 +13,27 @@
...
+ * mask representing the class that each pixel in the original image belong to.
  */
-public class SemanticSegmentationTranslator extends BaseImageTranslator<Image> {
+public class SemanticSegmentationTranslator extends BaseImageTranslator<Segmentation> {
 
     private final int shortEdge;
     private final int maxEdge;
 
-    private static final int CHANNEL = 3;
     private static final int CLASSNUM = 21;
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/modality/cv/translator/SemanticSegmentationTranslator.java:37]: This should comes from model, for deeplabv3, the classes can be found: https://github.com/pytorch/vision/blob/main/torchvision/models/_meta.py#L1128-L1150

File: api/src/main/java/ai/djl/modality/cv/translator/SemanticSegmentationTranslator.java [hunk 3]
```diff
@@ -57,69 +52,39 @@ public SemanticSegmentationTranslator(Builder builder) {
...
+
+        // Build mask array
+        for (int h = 0; h < height; h++) {
+            for (int w = 0; w < width; w++) {
+                int index = h * width + w;
+                int maxi = 0;
+                double maxnum = -Double.MAX_VALUE;
+                for (int i = 0; i < CLASSNUM; i++) {
+                    // get score for each i at the h,w pixel of the image
+                    float score = scores[i * (imageSize) + index];
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/modality/cv/translator/SemanticSegmentationTranslator.java:78]:
```suggestion
                    float score = scores[i * imageSize + index];
```

File: api/src/main/java/ai/djl/modality/cv/output/Segmentation.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * Constructs a Mask with the given data.
+     *
+     * @param mask the category mask for each pixel in the image
+     */
+    public Segmentation(int[][] mask) {
+        this.mask = mask;
+    }
+
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/modality/cv/output/Segmentation.java:50]: We need implement the following method:

public void drawMask(Image img, float transparency) {
...
}

public void drawMask(Image img, float transparency, int classId) {
...
}

public void drawMask(Image img, float transparency, int classId, Color color) {
...
}

public void Image extractImage(Image img, int classId) {
...
}

File: examples/src/main/java/ai/djl/examples/inference/SemanticSegmentation.java [hunk 5]
```diff
@@ -66,28 +66,18 @@ public static void predict() throws IOException, ModelException, TranslateExcept
...
-                Predictor<Image, Image> predictor = model.newPredictor()) {
-            Image semanticImage = predictor.predict(img);
-            saveSemanticImage(semanticImage);
+        try (ZooModel<Image, Segmentation> model = criteria.loadModel();
+                Predictor<Image, Segmentation> predictor = model.newPredictor()) {
+            return predictor.predict(img);
         }
     }
-
-    private static void saveSemanticImage(Image img) throws IOException {
```
Comment (frankfliu) [inline examples/src/main/java/ai/djl/examples/inference/SemanticSegmentation.java:85]: We still want to save segmented image, and we can save a few different mode:
1. only colors
2. transparent color overlay on existing image
3. replace background with other image

**Review by frankfliu — APPROVED** [review]

File: android/core/src/main/java/ai/djl/android/core/BitmapImageFactory.java [hunk 6]
```diff
@@ -120,6 +120,12 @@ public Image fromNDArray(NDArray array) {
         return new BitMapWrapper(bitmap);
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public Image fromArray(int[] pixels, int width, int height) {
```
Comment (frankfliu) [inline android/core/src/main/java/ai/djl/android/core/BitmapImageFactory.java:125]: Android depends on published snapshot build.
Move android changes into separate PR.

File: api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java [hunk 7]
```diff
@@ -157,6 +165,16 @@ public Object getWrappedImage() {
             return image;
         }
 
+        @Override
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java:168]:
```suggestion
        /** {@inheritDoc} */
        @Override
```

File: android/core/src/main/java/ai/djl/android/core/BitmapImageFactory.java (hunk 6 above)
Comment (xyang16) [reply android/core/src/main/java/ai/djl/android/core/BitmapImageFactory.java]: OK

File: api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java (hunk 7 above)
Comment (xyang16) [reply api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java:169]: Done

**Review by frankfliu — APPROVED** [review]
